documentation/OpenMP-4.5-grammar.txt

```diff
@@ -1,114 +1,396 @@
+        target-teams-distribute-simd-clause -> target-clause |
+                                               teams-distribute-simd-clause
+
+2.11.14 teams-distribute-parallel-do ->
+           TEAMS DISTRIBUTE PARALLEL DO [teams-distribute-parallel-do-clause[ [,] teams-distribute-parallel-do-clause]...]
+        teams-distribute-parallel-do-clause -> teams-clause |
+                                               distribute-parallel-do-clause
+
 2.11.15 target-teams-distribute-parallel-do ->
-                             TARGET TEAMS DISTRIBUTE PARALLEL DO
+           TARGET TEAMS DISTRIBUTE PARALLEL DO [target-teams-distribute-parallel-do-clause[ [,] target-teams-distribute-parallel-do-clause]...]
+        target-teams-distribute-parallel-do-clause -> target-clause |
+                                                      teams-distribute-parallel-do-clause
+
 2.11.16 teams-distribute-parallel-do-simd ->
-                             TEAMS DISTRIBUTE PARALLEL DO SIMD
+           TEAMS DISTRIBUTE PARALLEL DO SIMD [teams-distribute-parallel-do-simd-clause[ [,] teams-distribute-parallel-do-simd-clause]...]
+        teams-distribute-parallel-do-simd-clause -> teams-clause |
+                                                    distribute-parallel-do-simd-clause
+
 2.11.17 target-teams-distribute-parallel-do-simd ->
-                             TARGET TEAMS DISTRIBUTE PARALLEL DO SIMD
+           TARGET TEAMS DISTRIBUTE PARALLEL DO SIMD [target-teams-distribute-parallel-do-simd-clause[ [,] target-teams-distribute-parallel-do-simd-clause]...]
+        target-teams-distribute-parallel-do-simd-clause -> target-clause |
+                                                           teams-distribute-parallel-do-simd-clause
+
 2.13.1 master -> MASTER
-2.13.2 critical -> CRITICAL
+
+2.13.2 critical -> CRITICAL [(name) [HINT(hint-expr)]]
+
+2.13.2 end-critical -> END CRITICAL [(name)]
+
 2.13.3 barrier -> BARRIER
+
 2.13.4 taskwait -> TASKWAIT
+
 2.13.5 taskgroup -> TASKGROUP
-2.13.6 atomic -> ATOMIC
-2.13.7 flush -> FLUSH
-2.13.8 ordered -> ORDERED
-2.14.1 cancel -> CANCEL
-2.14.2 cancellation-point -> CANCELLATION POINT
-2.15.2 threadprivate -> THREADPRIVATE
-2.16 declare-reduction -> DECLARE REDUCTION
+
+2.13.6 atomic -> ATOMIC [seq_cst[,]] READ | WRITE | UPDATE | CAPTURE [[,]seq_cst]
+2.13.6 atomic -> ATOMIC [seq_cst]
+
+2.13.7 flush -> FLUSH [(variable-name-list)]
+
+2.13.8 ordered -> ORDERED ordered-construct-clause [[[,] ordered-construct-clause]...]
+       ordered-construct-clause -> depend-clause
+
+2.14.1 cancel -> CANCEL construct-type-clause [ [,] if-clause]
+       construct-type-clause -> PARALLEL |
+                                SECTIONS |
+                                DO |
+                                TASKGROUP
+
+2.14.2 cancellation-point -> CANCELLATION POINT construct-type-clause
+
+2.15.2 threadprivate -> THREADPRIVATE (variable-name-list)
+
```

Comment:
I would expect to see the productions for section 2.15.3 here.

Reply:
There is no production in Section 2.15.3. Clauses in 2.15.3.x were listed in the `Clauses` section at the end.